src/rdkafka.c

```diff
@@ -3495,24 +3499,11 @@ rd_kafka_poll_cb (rd_kafka_t *rk, rd_kafka_q_t *rkq, rd_kafka_op_t *rko,
 			rk->rk_conf.error_cb(rk, rko->rko_err,
 					     rko->rko_u.err.errstr,
                                              rk->rk_conf.opaque);
-                else {
-                        /* If error string already contains
-                         * the err2str then skip including err2str in
-                         * the printout */
-                        if (rko->rko_u.err.errstr &&
-                            strstr(rko->rko_u.err.errstr,
-                                   rd_kafka_err2str(rko->rko_err)))
-                                rd_kafka_log(rk, LOG_ERR, "ERROR",
-                                             "%s: %s",
-                                             rk->rk_name,
-                                             rko->rko_u.err.errstr);
-                        else
-                                rd_kafka_log(rk, LOG_ERR, "ERROR",
-                                             "%s: %s: %s",
-                                             rk->rk_name,
-                                             rko->rko_u.err.errstr,
-                                             rd_kafka_err2str(rko->rko_err));
-                }
+                else
```

Comment:
I think error events should be logged regardless of whether there is an `error_cb` since the behavior of `rd_kafka_log` is exactly what i'd want most of the time in response to one of these 'error' events, and the way the API is forces me to explicitly write code to do this if I set an `error_cb`.

Also, my memory is failing me on the specifics, but is `LOG_ERR` really appropriate for all these events? (don't have a good suggestion on what to do).

Reply:
I think the point of an error callback is for the application to handle the error in any way it seems fit, perhaps by logging it - in which case we would have duplicate logs.
If there is no error handler we log it instead so the error is not silently ignored.
It was evident in the examples/ programs that had error_cb that error+log was not great.

We can adjust this later if needed.